### PR TITLE
feat(SegmentedControl): new component

### LINKED ISSUE #242: [Feat]: New Segmented Control component
**Description**:

The `Segmented Control` component allows users to toggle between different views or settings, typically in a compact form where only one option is selected at a time. It is ideal for use in navigation, filtering, or mode-switching within apps and interfaces.

**Requirements**:

1. **Component Implementation:**

- Implement the `Segmented Control `component in TypeScript.
- Utilize SCSS with BEM methodology for styling.
- Integrate design tokens from @geneui/tokens for consistent colors, font sizes, spacing, and borders.
- Support different configurations such as single-line or multi-line options.
- Provide options to customize the appearance of active and inactive segments.
- Ensure accessibility with proper ARIA roles, focus management, and keyboard navigation.

2. **Unit Tests:**

- Write unit tests using Jest and React Testing Library.
- Ensure test coverage for all props, including rendering of different segment configurations and active states.
- Test for edge cases such as multiple selected segments, disabled segments, and rendering with long or truncated labels.

3. **Storybook Stories:**

- Create a Storybook story for the Segmented Control component.
- Showcase different configurations, such as single-line vs. multi-line segments, different segment states, and customization options.
- Create controls for the props so users can interact with the Segmented Control component in Storybook.
- Document all props of the public interfaces.

4. **Figma Design:**

Follow the design outlined in the provided Figma link.

**Checklist**:

- [ ]  Component is implemented in TypeScript.
- [ ]  Styling follows SCSS using BEM methodology.
- [ ]  Design tokens from @geneui/tokens are used for colors, font sizes, spacing, and borders.
- [ ]  Unit tests written and cover different states and edge cases.
- [ ]  Storybook stories created with examples of all props and configuration options.
- [ ]  Figma design matches the component implementation.

### PR DESCRIPTION
**Pull request by NarekArshakyan**
### Chromatic Build Summary 
**Build URL:** https://www.chromatic.com/build?appId=66daee7859a843dd87bfb3d7&number=875 
**Storybook URL:** https://66daee7859a843dd87bfb3d7-rdfityfbrx.chromatic.com/

### CODE REVIEW

File: src/components/atoms/SegmentedControl/SegmentedControl.tsx [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+import "./SegmentedControl.scss";
+
+// Component
+import { IconProps } from "@geneui/icons";
+import { HelperText, Label } from "../../../index";
+
+// Types
+import { IGlobalProps } from "./types";
+
+interface ISegmentedControlProps extends Omit<IGlobalProps, "name" | "children" | "isSelected"> {
```
[the remark this answers is not in the document]
Comment (hamikhambardzumyan) [reply src/components/atoms/SegmentedControl/SegmentedControl.tsx:20]: Here you don't need to extend another interface.

File: src/components/atoms/SegmentedControl/SegmentedControl.tsx [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+} from "react";
+// Styles
+import "./SegmentedControl.scss";
+
+// Component
+import { IconProps } from "@geneui/icons";
+import { HelperText, Label } from "../../../index";
+
+// Types
+import { IGlobalProps } from "./types";
```
Comment (hamikhambardzumyan) [inline src/components/atoms/SegmentedControl/SegmentedControl.tsx:18]:
```suggestion
```

File: src/components/atoms/SegmentedControl/SegmentedControl.tsx [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+interface ISegmentedControlProps extends Omit<IGlobalProps, "name" | "children" | "isSelected"> {
+    /**
+     * The text content of the `label`.
+     * This is the main text displayed within the `label`.
+     */
+    label?: string;
+    /**
+     * Control component. Renders inside the component
```
Comment (hamikhambardzumyan) [inline src/components/atoms/SegmentedControl/SegmentedControl.tsx:27]:
```suggestion
     * SegmentedControlItem component. Renders inside the component
```

File: src/components/atoms/SegmentedControl/SegmentedControl.tsx [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+     */
+    children: FunctionComponentElement<IGlobalProps> | FunctionComponentElement<IGlobalProps>[];
+    /**
+     * The actual text content to be displayed as helper text.
+     */
+    helperText?: string;
+    /**
+     * Works when the user clicks on one of the buttons. Returns the value that was written as a name in the Control component.
```
Comment (hamikhambardzumyan) [inline src/components/atoms/SegmentedControl/SegmentedControl.tsx:35]:
```suggestion
     * It works when the user clicks on one of the control items. Returns the value of the `name` prop from the `SegmentedControlItem`
```

File: src/components/atoms/SegmentedControl/SegmentedControl.tsx [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+     */
+    onChange: (name: string) => void;
+
+    // TODO:remove duplicate types after tests
+    /**
+     * The `Icon` prop accepts a React Functional Component that will be displayed alongside the button text.
+     */
+    Icon?: FC<IconProps>;
```
Comment (hamikhambardzumyan) [inline src/components/atoms/SegmentedControl/SegmentedControl.tsx:43]:
```suggestion
```

File: src/components/atoms/SegmentedControl/SegmentedControl.tsx [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+    /**
+     * Size <br>
+     * Possible values: `large | medium | small`
+     */
+    size?: "large" | "medium" | "small";
+
```
Comment (hamikhambardzumyan) [inline src/components/atoms/SegmentedControl/SegmentedControl.tsx:49]:
```suggestion
```

File: src/components/atoms/SegmentedControl/SegmentedControl.tsx [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+    /**
+     * Indicates whether the `button` is `disabled`, preventing user interaction, focus, click etc...
```
Comment (hamikhambardzumyan) [inline src/components/atoms/SegmentedControl/SegmentedControl.tsx:51]:
```suggestion
     * Indicates whether the `SegmentedControlItem` is `disabled`, preventing user interaction, focus, click etc...
```

File: src/components/atoms/SegmentedControl/SegmentedControl.tsx [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+     */
+    disabled?: boolean;
+    /**
+     *Displays the selected item
+     */
+    selected?: boolean;
+    /**
+     * Additional descriptive text shown with info icon and tooltip alongside of the label component.
+     */
+    infoText?: string;
```
Comment (hamikhambardzumyan) [inline src/components/atoms/SegmentedControl/SegmentedControl.tsx:61]: Move near to the `label` prop.

File: src/components/atoms/SegmentedControl/SegmentedControl.tsx [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+            e.preventDefault();
+            clickHandler(e);
+        }
+    };
+
+    const textSizes = props.size === "large" ? "medium" : props.size;
+
+    return (
+        <div className="segmentedControl">
+            <Label labelText={label} required size={textSizes} infoText={infoText} />
```
Comment (hamikhambardzumyan) [inline src/components/atoms/SegmentedControl/SegmentedControl.tsx:95]: Why is this always `required`? Add prop for required.

File: src/components/atoms/SegmentedControl/SegmentedControl.tsx [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+    };
+
+    const onKeyDown: React.KeyboardEventHandler<HTMLDivElement> = (e) => {
+        if (e.key === "Enter" || e.key === " ") {
+            e.preventDefault();
+            clickHandler(e);
+        }
+    };
+
+    const textSizes = props.size === "large" ? "medium" : props.size;
```
Comment (hamikhambardzumyan) [inline src/components/atoms/SegmentedControl/SegmentedControl.tsx:91]: Memoize this variable. 
Use destructuring to take the `size` prop

File: src/components/atoms/SegmentedControl/SegmentedControl.tsx [hunk 11]
```diff
@@ -0,0 +1,117 @@
...
+            setSelectedElementName(name);
+        }
+    };
+
+    const onKeyDown: React.KeyboardEventHandler<HTMLDivElement> = (e) => {
+        if (e.key === "Enter" || e.key === " ") {
+            e.preventDefault();
+            clickHandler(e);
+        }
+    };
```
Comment (hamikhambardzumyan) [inline src/components/atoms/SegmentedControl/SegmentedControl.tsx:89]:
```suggestion
```

File: src/components/atoms/SegmentedControl/SegmentedControl.tsx [hunk 12]
```diff
@@ -0,0 +1,117 @@
...
+    const clickHandler = (e: MouseEvent<HTMLDivElement> | KeyboardEvent<HTMLDivElement>) => {
+        e.stopPropagation();
+        const target = e.target as HTMLElement;
+        const name =
+            target.tagName === "BUTTON" ? target.getAttribute("name") : target.parentElement?.getAttribute("name");
+        if (name) {
+            onChange(name);
+            setSelectedElementName(name);
+        }
+    };
```
Comment (hamikhambardzumyan) [inline src/components/atoms/SegmentedControl/SegmentedControl.tsx:82]:
```suggestion
  const clickHandler = (name: string) => {
        setSelectedElementName(name);
    };
```

File: src/components/atoms/SegmentedControl/SegmentedControl.tsx [hunk 13]
```diff
@@ -0,0 +1,117 @@
...
+
```
Comment (hamikhambardzumyan) [inline src/components/atoms/SegmentedControl/SegmentedControl.tsx:83]: useEffect(() => {
        onChange(selectedElementName);
    }, [selectedElementName]);

File: src/components/atoms/SegmentedControl/SegmentedControl.tsx [hunk 14]
```diff
@@ -0,0 +1,117 @@
...
+                onKeyDown={onKeyDown}
+                tabIndex={-1}
+                aria-label="segmented control"
+                role="button"
+            >
+                {Children.map(children, (el) => {
+                    return cloneElement(el, {
+                        ...props,
+                        ...el.props,
+                        isSelected: selectedElementName ? selectedElementName === el.props.name : el.props.isSelected
```
Comment (hamikhambardzumyan) [inline src/components/atoms/SegmentedControl/SegmentedControl.tsx:108]:
```suggestion
                        selected: selectedElementName ? selectedElementName === el.props.name : el.props.selected,
                        onSelect: clickHandler
```

File: src/components/atoms/SegmentedControl/SegmentedControl.tsx [hunk 15]
```diff
@@ -0,0 +1,117 @@
...
+                className="segmentedControl__wrapper"
+                onClick={clickHandler}
+                onKeyDown={onKeyDown}
+                tabIndex={-1}
+                aria-label="segmented control"
+                role="button"
+            >
+                {Children.map(children, (el) => {
+                    return cloneElement(el, {
+                        ...props,
```
Comment (hamikhambardzumyan) [inline src/components/atoms/SegmentedControl/SegmentedControl.tsx:106]: Remove the `...props` and pass only props that need for the `SegmentedControlItem`.

File: src/components/atoms/SegmentedControl/SegmentedControl.tsx [hunk 16]
```diff
@@ -0,0 +1,117 @@
...
+
+    const textSizes = props.size === "large" ? "medium" : props.size;
+
+    return (
+        <div className="segmentedControl">
+            <Label labelText={label} required size={textSizes} infoText={infoText} />
+            <div
+                className="segmentedControl__wrapper"
+                onClick={clickHandler}
+                onKeyDown={onKeyDown}
```
Comment (hamikhambardzumyan) [inline src/components/atoms/SegmentedControl/SegmentedControl.tsx:99]:
```suggestion
```

File: src/components/atoms/SegmentedControl/SegmentedControl.tsx [hunk 17]
```diff
@@ -0,0 +1,117 @@
...
+                tabIndex={-1}
```
Comment (hamikhambardzumyan) [inline src/components/atoms/SegmentedControl/SegmentedControl.tsx:100]:
```suggestion
```

File: src/components/atoms/SegmentedControl/SegmentedControl.tsx [hunk 18]
```diff
@@ -0,0 +1,117 @@
...
+                aria-label="segmented control"
+                role="button"
```
Comment (hamikhambardzumyan) [inline src/components/atoms/SegmentedControl/SegmentedControl.tsx:102]:
```suggestion
```

File: src/components/atoms/SegmentedControl/Control.tsx [hunk 19]
```diff
@@ -0,0 +1,27 @@
+import React, { FC } from "react";
+import classNames from "classnames";
+
+// styles
+import "./SegmentedControl.scss";
+
+// types
+import { IGlobalProps } from "./types";
+
+const Control: FC<IGlobalProps> = ({ children, name, Icon, size = "medium", disabled = false, isSelected = false }) => {
```
Comment (hamikhambardzumyan) [inline src/components/atoms/SegmentedControl/Control.tsx:10]: Take the `onSelect` from props.

File: src/components/atoms/SegmentedControl/Control.tsx [hunk 20]
```diff
@@ -0,0 +1,27 @@
...
+    return (
+        <button
+            name={name}
```
Comment (hamikhambardzumyan) [inline src/components/atoms/SegmentedControl/Control.tsx:13]: Add click handler here.
```
onClick={() => onSelect(name)}
```

**Review by hamikhambardzumyan — CHANGES_REQUESTED** [review]

**Review by AregSoft — DISMISSED** [review]
